aws/resource_aws_service_discovery_http_namespace_test.go

```diff
@@ -0,0 +1,92 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/servicediscovery"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSServiceDiscoveryHttpNamespace_basic(t *testing.T) {
+	resourceName := "aws_service_discovery_http_namespace.test"
+	rName := "terraformtesting"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsServiceDiscoveryHttpNamespaceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccServiceDiscoveryHttpNamespaceConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsServiceDiscoveryHttpNamespaceExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "arn"),
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "description", "test"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAwsServiceDiscoveryHttpNamespaceDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).sdconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_service_discovery_http_namespace" {
+			continue
+		}
+
+		input := &servicediscovery.GetNamespaceInput{
+			Id: aws.String(rs.Primary.ID),
+		}
+
+		_, err := conn.GetNamespace(input)
+		if err != nil {
+			if isAWSErr(err, servicediscovery.ErrCodeNamespaceNotFound, "") {
+				return nil
```

Comment:
This `return nil` should be replaced with `continue` so the for loop can check other `aws_service_discovery_http_namespace` resources for deletion if/when a test configuration contains multiple. 👍 

```suggestion
				continue
```